### PR TITLE
feat: block sycner add txhash when export events & juno version update

### PR DESCRIPTION
**Pull request by krish-nr**
### Description

block syncer add txhash info when export events. Also update juno version for stop serving function and include SP module  and GC function. 

### Rationale

1. We need txhash info when seal an object. Block syncer add this context.
2. Chain side add the new feat of stop serving, we implement that in juno and here with juno version updated
3. block syncer persist SP metadata by handling SP events in juno
4. adapt chain side changes

### Example

N/A

### Changes

Notable changes: 
* add txhash when export events
* adapt indexer interface change

### CODE REVIEW

File: service/blocksyncer/block_syncer_indexer.go [hunk 1]
```diff
@@ -61,24 +62,45 @@ func (i *Impl) HandleEvent(ctx context.Context, block *coretypes.ResultBlock, tx
...
 	block, err := i.Node.Block(int64(height))
 	if err != nil {
 		log.Errorf("failed to get block from node: %s", err)
 		return err
 	}
 
+	// get txs
+	txs, err := i.Node.Txs(block)
+	if err != nil {
+		return fmt.Errorf("failed to get transactions for block: %s", err)
```
Comment (sysvm) [inline service/blocksyncer/block_syncer_indexer.go:75]: define an error and add error log

File: service/blocksyncer/block_syncer_indexer.go [hunk 2]
```diff
@@ -61,24 +62,45 @@ func (i *Impl) HandleEvent(ctx context.Context, block *coretypes.ResultBlock, tx
...
+
+	if len(beginBlockEvents) > 0 {
+		i.ExportEventsWithoutTx(context.Background(), block, beginBlockEvents)
+	}
 
-	err = i.ExportEvents(context.Background(), block, events)
+	// 2. handle events in txs
+	err = i.ExportEventsInTxs(context.Background(), block, txs)
 	if err != nil {
-		log.Errorf("failed to ExportEvents: %s", err)
```
Comment (sysvm) [inline service/blocksyncer/block_syncer_indexer.go:78]: why not add log

File: service/blocksyncer/block_syncer_indexer.go [hunk 3]
```diff
@@ -3,6 +3,7 @@ package blocksyncer
 import (
 	"context"
 	"encoding/json"
+	"fmt"
 
 	"github.com/bnb-chain/greenfield-storage-provider/pkg/log"
 	"github.com/bnb-chain/greenfield-storage-provider/pkg/metrics"
```
Comment (sysvm) [inline service/blocksyncer/block_syncer_indexer.go:9]: pkg group as follows: std lib, third-party, inner lib

**Review by sysvm — APPROVED** [review]
LGTM

**Review by constwz — APPROVED** [review]

**Pull request merged**